internal/build/imgsrc/resolver.go

```diff
@@ -46,6 +49,52 @@ type ImageOptions struct {
 	Label           map[string]string
 }
 
+func (io ImageOptions) ToSpanAttributes() []attribute.KeyValue {
+	attrs := []attribute.KeyValue{
+		attribute.String("imageoptions.app_name", io.AppName),
+		attribute.String("imageoptions.work_dir", io.WorkingDir),
+		attribute.String("imageoptions.dockerfile_path", io.DockerfilePath),
+		attribute.String("imageoptions.ignorefile_path", io.IgnorefilePath),
+		// todo(gwuah): find out from security if this is fine
+		attribute.String("imageoptions.image.ref", io.ImageRef),
+		attribute.String("imageoptions.image.label", io.ImageLabel),
+		attribute.Bool("imageoptions.publish", io.Publish),
+		attribute.String("imageoptions.tag", io.Tag),
+		attribute.Bool("imageoptions.nocache", io.NoCache),
+		attribute.String("imageoptions.builtin", io.BuiltIn),
+		attribute.String("imageoptions.builder", io.BuiltIn),
+		attribute.StringSlice("imageoptions.buildpacks", io.Buildpacks),
+	}
+
+	b, err := json.Marshal(io.BuildArgs)
+	if err == nil {
+		attrs = append(attrs, attribute.String("imageoptions.buildArgs", string(b)))
+	}
+
+	b, err = json.Marshal(io.ExtraBuildArgs)
+	if err == nil {
+		attrs = append(attrs, attribute.String("imageoptions.extraBuildArgs", string(b)))
+	}
+
+	b, err = json.Marshal(io.BuildSecrets)
+	if err == nil {
+		attrs = append(attrs, attribute.String("imageoptions.buildSecrets", string(b)))
+	}
+
+	b, err = json.Marshal(io.BuiltInSettings)
+	if err == nil {
+		attrs = append(attrs, attribute.String("imageoptions.builtInSettings", string(b)))
```

Comment:
Some of the attributes in this diff are snake case and some camel case. What's the preferred form? Do they need to be consistent?